src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -21,6 +21,7 @@ namespace System.Runtime.CompilerServices
     internal static unsafe class CastHelpers
     {
         private static int[]? s_table;
+        private static readonly MethodTable* pObjMt = RuntimeHelpers.GetMethodTable(new object());
```

Comment:
Would it be better to use `GetValueInternal` intrinsic instead?

Reply:
I did not think about using `GetValueInternal` here. I guess, why not. It would skip an allocation.